Add warnings about NFS (#7691)
prometheus/prometheus: docs/storage.md

To tune the rate of ingested samples per second, you can either reduce the number of time series you scrape (fewer targets or fewer series per target), or you can increase the scrape interval. However, reducing the number of series is likely more effective, due to compression of samples within a series.

If your local storage becomes corrupted for whatever reason, your best bet is to shut down Prometheus and remove the entire storage directory. You can try removing individual block directories, or WAL directory, to resolve the problem, this means losing a time window of around two hours worth of data per block directory. Again, Prometheus's local storage is not meant as durable long-term storage.

> CAUTION: Non POSIX compliant filesystems are not supported by Prometheus's local storage, corruptions may happen, without possibility to recover. NFS is only potentially POSIX, most implementations are not. Cases of corruption are also reported with other networked filesystems. It is strongly recommended to use a local filesystem for reliability.

Reviewer: ```suggestion
CAUTION: Non-POSIX compliant filesystems are not supported by Prometheus' local storage as unrecoverable corruptions may happen. NFS could be POSIX-compliant, but most implementations are not. Cases of corruption are also reported with other networked filesystems. It is strongly recommended to use a local filesystem for reliability.
```

> CAUTION: Non-POSIX compliant filesystems are not supported by Prometheus' local storage as unrecoverable corruptions may happen. NFS could be POSIX-compliant, but most implementations are not. Notably, AWS EFS is known to cause corruptions. Cases of corruption are also reported with other networked filesystems. It is strongly recommended to use a local filesystem for reliability.

Reviewer 2: That's a bit harsh on EFS - we don't have definitive proof but there seems to be a fair few users using it who have had corruption. I'd weaken to something like "NFS filesystems (including AWS's EFS) are not supported".

> Cases of corruption are also reported with other networked filesystems.

It's not yet clear if SMB is an issue, so I'd leave this out for now.

Author: We do have a report today about glusterfs.

Author: (on the mailing list I think)

Author: I have removed the sentence. We can still adapt later.

If your local storage becomes corrupted for whatever reason, your best bet is to shut down Prometheus and remove the entire storage directory. You can try removing individual block directories, or WAL directory, to resolve the problem, this means losing a time window of around two hours worth of data per block directory. Again, Prometheus's local storage is not meant as durable long-term storage.

> CAUTION: Non-POSIX compliant filesystems are not supported by Prometheus' local storage as unrecoverable corruptions may happen. NFS could be POSIX-compliant, but most implementations are not. Cases of corruption are also reported with other networked filesystems. It is strongly recommended to use a local filesystem for reliability.

Reviewer 2: Should we explicitly also call out AWS EFS, as many users of it don't seem to realise it's NFS?

Author: Yes, noted.
